# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/tests/browser/fission/browser_take_focus.js
# Commit: 4572c1ba670a
# Full Hash: 4572c1ba670a3ddb159cf37ba9684fb6055f260e
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2021-01-30 05:43:48
# Regressor Bug: 1688972
# File Overlap Count: 1
# Description:
#   Bug 1688972 - Make FocusedChild work across document/process boundaries. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D103102
# ==============================================================================

diff -r f262a5526f6c -r 4572c1ba670a accessible/tests/browser/fission/browser_take_focus.js
--- a/accessible/tests/browser/fission/browser_take_focus.js	Fri Jan 29 22:44:30 2021 +0200
+++ b/accessible/tests/browser/fission/browser_take_focus.js	Fri Jan 29 20:57:30 2021 +0000
@@ -11,9 +11,15 @@
 );
 
 addAccessibleTask(
-  `<input id="textbox" value="hello"/>`,
+  `<div role="group"><input id="textbox" value="hello"/></div>`,
   async function(browser, iframeDocAcc, contentDocAcc) {
     const textbox = findAccessibleChildByID(iframeDocAcc, "textbox");
+    const iframe = findAccessibleChildByID(contentDocAcc, "default-iframe-id");
+    const iframeDoc = findAccessibleChildByID(
+      contentDocAcc,
+      "default-iframe-body-id"
+    );
+
     testStates(textbox, STATE_FOCUSABLE, 0, STATE_FOCUSED);
 
     let onFocus = waitForEvent(EVENT_FOCUS, textbox);
@@ -21,6 +27,35 @@
     await onFocus;
 
     testStates(textbox, STATE_FOCUSABLE | STATE_FOCUSED, 0);
+
+    is(
+      getAccessibleDOMNodeID(contentDocAcc.focusedChild),
+      "textbox",
+      "correct focusedChild from top doc"
+    );
+
+    is(
+      getAccessibleDOMNodeID(iframeDocAcc.focusedChild),
+      "textbox",
+      "correct focusedChild from child doc"
+    );
+
+    ok(!iframe.focusedChild, "correct focusedChild from iframe (null)");
+
+    onFocus = waitForEvent(EVENT_FOCUS, iframeDoc);
+    iframeDoc.takeFocus();
+    await onFocus;
+
+    is(
+      getAccessibleDOMNodeID(contentDocAcc.focusedChild),
+      "default-iframe-body-id",
+      "correct focusedChild of child doc from top doc"
+    );
+    is(
+      getAccessibleDOMNodeID(iframe.focusedChild),
+      "default-iframe-body-id",
+      "correct focusedChild of child doc from iframe"
+    );
   },
   { topLevel: false, iframe: true, remoteIframe: true }
 );
